modules/src/mock/context.rs

```diff
@@ -362,20 +382,30 @@ impl ChannelReader for MockContext {
         ClientReader::consensus_state(self, client_id, height)
     }
 
-    fn authenticated_capability(&self, port_id: &PortId) -> Result<Capability, ICS4Error> {
+    fn authenticated_capability(&self, port_id: &PortId) -> Result<Capability, Ics4Error> {
         let cap = PortReader::lookup_module_by_port(self, port_id);
         match cap {
             Some(key) => {
                 if !PortReader::authenticate(self, &key, port_id) {
-                    Err(ICS4Error::from(ICS4Kind::InvalidPortCapability))
+                    Err(Ics4Error::from(Ics4Kind::InvalidPortCapability))
                 } else {
                     Ok(key)
                 }
             }
-            None => Err(ICS4Error::from(ICS4Kind::NoPortCapability(port_id.clone()))),
+            None => Err(Ics4Error::from(Ics4Kind::NoPortCapability(port_id.clone()))),
         }
     }
 
+    fn get_next_sequence_send(&self, port_channel_id: &(PortId, ChannelId)) -> Option<&u64> {
+        self.next_sequence_send.get(port_channel_id)
+    }
+
+    fn hashing(&self, input: String) -> String {
```

Comment:
@romac I think this is where the new `rust-crypto` dependency is being used. A hash/digest is necessary when we call `store_packet_commitment`.

That being said, since we only use this method in the mock for tests, it would make sense to make this crate a dep only for `[dev-dependencies]`.

@cezarad: I would give this method a better name (usually methods in the Reader and Keeper are actions/verbs):

```suggestion
    fn hash(&self, input: String) -> String {
```